test/common/upstream/upstream_impl_test.cc

```diff
@@ -2332,20 +2389,50 @@ TEST_F(ClusterInfoImplTest, TestTrackTimeoutBudgets) {
   // The stats will be null if they have not been explicitly turned on.
   EXPECT_FALSE(cluster->info()->timeoutBudgetStats().has_value());
 
+  const std::string yaml_disabled2 = R"EOF(
+    name: name
+    connect_timeout: 0.25s
+    type: STRICT_DNS
+    lb_policy: ROUND_ROBIN
+    track_cluster_stats: { request_response_sizes : true }
+  )EOF";
+
+  cluster = makeCluster(yaml_disabled2);
+  EXPECT_FALSE(cluster->info()->timeoutBudgetStats().has_value());
+
+  const std::string yaml_disabled3 = R"EOF(
+    name: name
+    connect_timeout: 0.25s
+    type: STRICT_DNS
+    lb_policy: ROUND_ROBIN
+    track_cluster_stats: { timeout_budgets : false }
+  )EOF";
+
+  cluster = makeCluster(yaml_disabled3);
+  EXPECT_FALSE(cluster->info()->timeoutBudgetStats().has_value());
+}
+
+TEST_F(ClusterInfoImplTest, TestTrackTimeoutBudgets) {
   // Check that with the flag, the histogram is created.
   const std::string yaml = R"EOF(
     name: name
     connect_timeout: 0.25s
     type: STRICT_DNS
     lb_policy: ROUND_ROBIN
-    track_timeout_budgets: true
+    track_cluster_stats: { timeout_budgets : true }
   )EOF";
 
-  cluster = makeCluster(yaml);
+  auto cluster = makeCluster(yaml);
   // The stats should be created.
   EXPECT_TRUE(cluster->info()->timeoutBudgetStats().has_value());
+  EXPECT_EQ(
+      Stats::Histogram::Unit::Unspecified,
+      cluster->info()->timeoutBudgetStats()->get().upstream_rq_timeout_budget_percent_used_.unit());
```

Comment:
consider a temp for `cluster->info()->timeoutBudgetStats()->get().upstream_rq_timeout_budget_percent_used_` or similar.